File: app/src/main/java/mobi/moop/features/login/LoginActivity.java
package mobi.moop.features.login;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import butterknife.BindView;
import butterknife.ButterKnife;
import mobi.moop.R;

public class LoginActivity extends AppCompatActivity {


    @BindView(R.id.toolbarLogin)
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        ButterKnife.bind(this);
        setSupportActionBar(toolbar);
        toolbar.setNavigationIcon(R.drawable.ic_arrow_back);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
        showLoginFragment();

    }

    private void showLoginFragment() {
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.placeholder, new LoginFragment());
        ft.commit();
    }

    public void showRegistroFragment() {
        FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
        ft.replace(R.id.placeholder, new RegistroFragment());
        ft.addToBackStack("main");
        ft.commit();
    }

    public void hideToolbar() {
        toolbar.setVisibility(View.GONE);
    }

    public void showToolbar() {
        toolbar.setVisibility(View.VISIBLE);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
    }


}

File: app/src/main/java/brmobi/moop/ui/login/LoginActivity.kt
package brmobi.moop.ui.login

import android.content.Intent
import android.os.Bundle
import android.view.View
import brmobi.moop.R
import brmobi.moop.ui.base.BaseActivity
import kotlinx.android.synthetic.main.activity_login.*
import javax.inject.Inject

class LoginActivity : BaseActivity(), LoginActivityMpView {

    @Inject
    lateinit var mPresenter: LoginActivityMvpPresenter<LoginActivityMpView>

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        getActivityComponent().inject(this)
        mPresenter.onAttach(this)
        setSupportActionBar(toolbarLogin)
        toolbarLogin.setNavigationIcon(R.drawable.ic_arrow_back)
        toolbarLogin.setNavigationOnClickListener { onBackPressed() }
        showLoginFragment()

    }
    override fun onDestroy() {
        super.onDestroy()
        mPresenter.onDetach()
    }

    private fun showLoginFragment() {
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.placeholder, LoginFragment())
        ft.commit()
    }

    fun showRegistroFragment() {
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.placeholder, RegistroFragment())
        ft.addToBackStack("main")
        ft.commit()
    }

    fun hideToolbar() {
        toolbarLogin.visibility = View.GONE
    }

    fun showToolbar() {
        toolbarLogin.visibility = View.VISIBLE
    }


}
